pkg/cmd/boot/boot.go

```diff
@@ -505,7 +505,7 @@ func existingBootClone(dir string) (bool, error) {
 
 func (o *BootOptions) checkIfProvidedRequirementsArePossiblyStale() error {
 	_, devEnv := o.GetDevEnv()
-	if devEnv != nil {
```

Comment:
Took a look at the function, and you are right, it never returns nil. 
I think we should to return nil from GetDevEnv(), instead of &jenkinsv1.Environment{}. However this function is used in a lot of different places, so it might be a breaking change. wdyt @abayer?

Reply:
This is called in enough places that it won't be trivial to switch it to returning nil, but it's probably worth the attempt.

Reply:
I can give it a shot.

Reply:
That would be awesome :100:

Reply:
I've updated my PR to reflect this `GetDevEnv` returning `nil` instead.